txp/Pos/Txp/Toil/Utxo/Functions.hs

```diff
@@ -147,28 +145,25 @@ verifyConsistency inputs witnesses
     errFmt = ("length of inputs != length of witnesses "%"("%int%" != "%int%")")
     errMsg = sformat errFmt (length inputs) (length witnesses)
 
-verifyOutputs :: VTxContext -> TxAux -> VerificationRes
+verifyOutputs :: 
+       MonadError ToilVerFailure m 
+    => VTxContext
+    -> TxAux 
+    -> m ()
```

Comment:
Oh, previous version (with concrete `Either` monad instead of abstract `m` monad) was better.
Please see [exception handling guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md) and [monadic effects guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/monads.md).
Two points:
1. We don't write code in abstract `m` which is `MonadError`, because this code may be potentially impure and then there will be two ways of throwing exceptions.
2. We write [pure code](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/monads.md#pure-code) in concrete monads.

Reply:
Thanks, I was trying to copy other `verify` functions but you're right, the previous (concrete) version is better